skipruntime-ts/ts/src/skipruntime_api.ts

```diff
@@ -13,7 +13,14 @@ export type JSONObject = { [key: string]: TJSON | null };
 
 export type TJSON = number | JSONObject | boolean | (TJSON | null)[] | string;
 
-export type Param = any;
+export type Param =
+  | null
+  | boolean
+  | number
+  | string
+  | Frozen
+  | readonly Param[]
+  | { readonly [k: string]: Param };
```

Comment:
Do we want to allow arrays and objects that are not themselves `Frozen` but which might eventually contain `Frozen` data? It would be more conservative to, at least initially, require arrays and objects to be `Frozen`. I don't trust `readonly` in TS to achieve anything. Also, IIUC the `check` function will blow at runtime on non-frozen arrays and objects.

Reply:
The goal was not to make `check` always pass but start with reducing the cases where it would blow.

Currently `check` accepts more things than values returned by `freeze`, it accepts values that are manually deeply frozen.

Reply:
Ah yes, check will accept some manually deep-frozen values. But it is slow in that case as it has to do a deep traversal every time. I seem to recall, but can't currently find, some discussion where check accepting in these cases was an intermediate state but not what we want eventually. Do you recall @skiplabsdaniel ?